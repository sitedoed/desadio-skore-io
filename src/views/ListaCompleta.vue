<template>
    <div class="wrapper">
        <div class="container">
            <div class="item">
                <section>
                    <h1>{{ page_title }}</h1>
                    <p>Lista simples de contéudos, com seus respectivos campos, para referência.
                        Os campos "vazios" não têm valor cadastrado, como pode ser observado non link:
                        <strong>Saiba mais</strong>
                        de cada conteúdo. 
                    </p>
                    <br />
                </section>
                <section>
                    <article
                        v-for="item in contents"
                        :key="item.id"
                        class=""
                    >
                        <h3 class="">{{ item.title }}</h3>
                        <div class="">
                        <p><strong>Descrição: </strong> {{ item.description }}</p>
                        <ul>
                            <li><strong>Tipo: </strong> {{ item.type }}</li>
                            <li class="hide"><strong>Url:</strong> {{ item.url }}</li>
                            <li><strong>Embeddable:</strong> {{ item.embeddable }}</li>
                            <li><strong>Download:</strong> {{ item.allow_download }}</li>
                            <li><strong>Data de criação:</strong> {{ item.created_at }}</li>
                            <li><strong>Última atualização:</strong> {{ item.updated_at }}</li>
                            <li><router-link :to="`/saiba_mais/${item.id}`"><span class="saiba-mais">Saiba mais</span></router-link></li>
                        </ul>
                        </div>
                    </article>
                </section>
            
            </div>
        </div>
    </div>
</template>

<script>
import gql from 'graphql-tag'
import '../assets/css/main.css'

export default {
    name: "ListaCompletaVue",
    data: function() {
        return {
        page_title: "Lista Completa"
        };
    },
    apollo: {
    contents: gql `
    query contents {
        contents {
        id
        title
        description
        type
        url
        embeddable
        allow_download
        created_at
        updated_at
        }
    }
    `,
    },
}
</script>

<style>
.bold{
    font-weight:bold ;
}
</style>