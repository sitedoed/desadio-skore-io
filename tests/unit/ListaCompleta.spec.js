import { createLocalVue, shallowMount } from '@vue/test-utils';
import ListaCompleta from "@/views/ListaCompleta.vue";

describe("Testing ListaCompletaVue Component", () => {

    it("Esta exibindo o nome da página?", () => {
        const wrapper = shallowMount(ListaCompleta, {
            props: {
                page_title: "Lista Completa"
            }
        })
    })
    it('Está carregando a lista?', () => {
      const wrapper = shallowMount(ListaCompleta, {
        mocks: {
          $apollo: {
            queries: {
              contents: {
                loading: true,
              },
            },
          },
        },
      })
      expect(wrapper.element).toMatchSnapshot();
    })
      
})
