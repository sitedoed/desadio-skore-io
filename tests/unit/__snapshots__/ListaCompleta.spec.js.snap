// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ListaCompletaVue Component Está carregando a lista? 1`] = `
<div
  class="wrapper"
>
  <div
    class="container"
  >
    <div
      class="item"
    >
      <section>
        <h1>
          Lista Completa
        </h1>
        <p>
          Lista simples de contéudos, com seus respectivos campos, para referência. Os campos "vazios" não têm valor cadastrado, como pode ser observado non link: 
          <strong>
            Saiba mais
          </strong>
           de cada conteúdo. 
        </p>
        <br />
      </section>
      <section>
        
        
      </section>
    </div>
  </div>
</div>
`;
